pkg/share/manager/jsoncs3/jsoncs3.go

```diff
@@ -699,14 +694,60 @@ func (m *Manager) listCreatedShares(ctx context.Context, user *userv1beta1.User,
 	return ss, nil
 }
 
+// call in locked context (write lock)
+func (m *Manager) invalidateCache(ctx context.Context, user *userv1beta1.User) {
```

Comment:
Can we run `invalidateCache` in a seperate go-routine? (or at least parts of it?) As far as I see we don't need it for the request.

Reply:
Would not recommend with that implementation, I fear this will cause deadlocks and zombie processes with the Write-Semaphores. Suggestion => Move to a cache that properly deals with threading.